core/dive.h

```diff
@@ -759,10 +759,15 @@ extern void average_max_depth(struct diveplan *dive, int *avg_depth, int *max_de
 #ifdef __cplusplus
 }
 
-/* Make pointers to dive and dive_trip "Qt metatypes" so that they can
- * be passed through QVariants. */
+/* Make pointers to dive, dive_trip and dive_table "Qt metatypes" so that they can
+ * be passed through QVariants and through QML.
+ * Note: we have to use a typedef for "struct dive_table", because MOC removes the
+ * "struct", but dive_table is already the name of a global variable, leading to
+ * compilation errors. */
+typedef struct dive_table *dive_table_ptr;
 Q_DECLARE_METATYPE(struct dive *);
 Q_DECLARE_METATYPE(struct dive_trip *);
+Q_DECLARE_METATYPE(dive_table_ptr);
```

Comment:
btw, why does this work with `struct dive *` above? That seems inconsistent with what you said in the commit message.
Just curious

Reply:
> btw, why does this work with struct dive * above? 

Because there is no global variable with the name `dive`, but there is a global variable with the name `dive_table`.

What happens is that MOC declares parameters as `dive_table *xyz`, which is perfectly fine C++. Unless there is a variable with the name `dive_table`, which then takes precedence, so `dive_table *xyz` gets parsed as "multiply the object `dive_table` with `xyz`. Apparently the Qt developers didn't have classical C code in mind, where you can't leave out the struct and therefore variables with the same name as structs are rather common.

Reply:
BTW: I always thought that it was odd that Linus decided against typedefs for some of the structs. He had his logic (typedefs were for "simple" structures that were ok to do, say, assignments on... but given the way we use structs and of course given all the C++ influx, this really doesn't make sense anymore...